File: dotnet/cafdemalihapa/Directories/Applications/ApplicationLocation.cs
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Directories.Applications
{
    public class ApplicationLocation(
        ILogger<ApplicationLocation> logger
    )
    {
        public string GetPath()
        {
            var applicationLocation = CommandLineArgs.GetByKey("--application-location");
            logger.LogInformation("application location: {applicationLocation}", applicationLocation);
            return applicationLocation;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Applications/ApplicationLocation.fs
namespace cafdemwimu.console.Directories.Applications

open Microsoft.Extensions.Logging
open cafdemwimu.console

type ApplicationLocation(logger: ILogger<ApplicationLocation>) =
    member _.GetPath() =
        let applicationLocation = CommandLineArgs.GetByKey("--application-location")
        logger.LogInformation("application location: {applicationLocation}", applicationLocation)
        applicationLocation
